File: leetcode-go/100.go
package main

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p == nil && q == nil {
		return true
	}
	if (p == nil && q != nil) || (p != nil && q == nil) || (p != nil && q != nil && p.Val != q.Val) {
		return false
	}
	traversalResult := isSameTree(p.Left, q.Left)
	traversalResult = traversalResult && isSameTree(p.Right, q.Right)
	return traversalResult
}

File: leetcode-js/100.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {
    let result = true;

    const compareTrees = (p, q) => {
        if (!p && !q) return;
        if ((!p && q) || (p && !q) || (p?.val !== q?.val)) {
            result = false;
            return;
        }
        compareTrees(p.left, q.left);
        compareTrees(p.right, q.right);
    }
    compareTrees(p, q);
    return result;
};